errors.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//     http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package goacme
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+)
+
+// ErrorType is used to define a set of errors predefined by the ACME spec.
+type ErrorType string
+
+const (
+	ErrBadCSR       = "urn:acme:error:badCSR"         // The CSR is unacceptable (e.g., due to a short key)
```

Comment:
Yes, exactly. Const errors are normally defined as e.g. `errors.New("my error")`. For instance, that's precisely how [io.EOF](https://golang.org/pkg/io/#pkg-variables) and other io errors are defined.

In this case though we also have `detail` field (free text string) which may vary and cannot be a constant as the actual value is defined by an CA server response that we don't know a priori, so we can't have const errors unless `detail` field is removed or replaced with a const value in this package which may remove important information when propagating error messages from a CA server.

Thoughts?